v1/tarball/image.go

```diff
@@ -104,11 +141,13 @@ func (i *image) loadManifest() error {
 	if err := json.Unmarshal(mfstBytes, &tarMfst); err != nil {
 		return err
 	}
-	if len(tarMfst) != 1 {
-		return errors.New("tarball must contain only a single image to be used with tarball.Image")
+
+	mfst, err := findSpecifiedManifest(i.tag, tarMfst)
+	if err != nil {
+		return err
 	}
 
-	cfgBytes, err := extractFileFromTar(i.path, tarMfst[0].Config)
+	cfgBytes, err := extractFileFromTar(i.path, mfst.Config)
```

Comment:
Ok, so here's the nuance of the laziness I was talking about :)
1. We need to read `manifest.json` to access *everything* (might as well be eager), however,
1. We do NOT need `i.manifest` computed eagerly (this requires compressing all of the layers, which is expensive).